// react-simple-file-upload/dist/index.js
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var React = require('react');
var React__default = _interopDefault(React);
var shortid = _interopDefault(require('shortid'));
var makeDebug = _interopDefault(require('debug'));

var SimpleFileUploadContext = React.createContext({
  apiKey: '',
  width: 160,
  height: 160
});
var useSimpleFileUpload = function useSimpleFileUpload() {
  return React.useContext(SimpleFileUploadContext);
};
var SimpleFileUploadProvider = function SimpleFileUploadProvider(_ref) {
  var apiKey = _ref.apiKey,
      width = _ref.width,
      height = _ref.height,
      children = _ref.children;
  var value = {
    apiKey: apiKey,
    width: +width,
    height: +height
  };
  return /*#__PURE__*/React__default.createElement(SimpleFileUploadContext.Provider, {
    value: value
  }, children);
};
var debug = makeDebug('SimpleFileUpload');

var SimpleFileUpload = function SimpleFileUpload(_ref2) {
  var apiKey = _ref2.apiKey,
      onSuccess = _ref2.onSuccess,
      width = _ref2.width,
      height = _ref2.height,
      preview = _ref2.preview,
      text = _ref2.text;
  var sfu = useSimpleFileUpload();
  var key = sfu.apiKey || apiKey;
  width = width || sfu.width;
  height = height || sfu.height;
  text = text || "Drop file to upload";
  var small = "false";

  if (parseInt(width) < 120) {
    small = "true";
  }

  var widgetId = React.useRef(shortid.generate());
  React.useEffect(function () {
    window.addEventListener('message', handleIframeMessage, false);
    return function () {
      window.removeEventListener('message', handleIframeMessage);
    };
  });

  var handleIframeMessage = function handleIframeMessage(e) {
    if (e.origin !== 'https://app.simplefileupload.com') {
      return;
    }

    if (e.data.uploadResult === 'success') {
      if (e.data.widgetId !== widgetId.current) {
        debug('Ignoring because widgetId does not match');
        return;
      }

      if (e.data.widgetId === widgetId.current) {
        debug('Success: %O', e.data);
        onSuccess(e.data.url);
      }
    }
  };

  return /*#__PURE__*/React__default.createElement("iframe", {
    title: "Simple File Upload " + widgetId.current,
    src: "https://app.simplefileupload.com/buckets/" + key + "?widgetId=" + widgetId.current + "&preview=" + preview + "&text=" + text + "&small=" + small,
    className: "widgetFrame",
    width: width,
    height: height,
    style: {
      border: 0
    }
  });
};

exports.SimpleFileUploadContext = SimpleFileUploadContext;
exports.SimpleFileUploadProvider = SimpleFileUploadProvider;
exports.default = SimpleFileUpload;
exports.useSimpleFileUpload = useSimpleFileUpload;
